maap/dps/dps_job.py

```diff
@@ -124,18 +99,12 @@ def retrieve_attributes(self):
                 self.retrieve_metrics()
             except:
                 pass
+        return self
 
     def cancel_job(self):
-        url = os.path.join(MaapConfig().dps_job, endpoints.DPS_JOB_DISMISS, self.id)
-        headers = RequestsUtils.generate_dps_headers()
-        logger.debug('DELETE request sent to {}'.format(url))
-        logger.debug('headers:')
-        logger.debug(headers)
-        response = requests.post(
-            url=url,
-            headers=headers
-        )
-        return RequestsUtils.check_response(response)
+        url = os.path.join(self.config.dps_job, endpoints.DPS_JOB_DISMISS, self.id)
```

Comment:
Use urllib methods for constructing URLs, no os.path methods since os.path methods have OS-dependent results (I realize this is not likely to be run on a Windows platform, but this is still a good habit). There are a few other spots where you should make such a change, but I'll leave that for you to address instead of making multiple comments.

```suggestion
        url = urllib.parse.join(self.config.dps_job, f"{endpoints.DPS_JOB_DISMISS}/{self.id}")
```